packages/block-editor/src/components/block-list/insertion-point.js

```diff
@@ -200,6 +200,18 @@ function InsertionPointPopover( {
 		}
 	}
 
+	// Only show the inserter when there's a `nextElement` (a block after the
+	// insertion point). At the end of the block list the trailing appender
+	// should serve the purpose of inserting blocks.
```

Comment:
What if there's no appender available?

Reply:
I don't think that's an issue, from what I understand this inserter was never intended to show at the end of a block list. Its appearance there is a bug from https://github.com/WordPress/gutenberg/pull/28418.